themes/ddbasic/template.node.php

```diff
@@ -0,0 +1,369 @@
+    else {
+      // User OG group ref to link back to library.
+      if (isset($variables['content']['og_group_ref'])) {
+        $variables['ddbasic_event_location'] = $variables['content']['og_group_ref'];
+      }
+    }
+  }
+
+  // Add tpl suggestions for node view modes.
+  if (isset($variables['view_mode'])) {
+    $variables['theme_hook_suggestions'][] = 'node__view_mode__' . $variables['view_mode'];
+  }
+
+  // For search result view mode move title into left col. group.
+  if (isset($variables['content']['group_right_col_search'])) {
+    $variables['content']['group_right_col_search']['title'] = array(
+      '#theme' => 'link',
+      '#text' => decode_entities($variables['title']),
+      '#path' => 'node/' . $variables['nid'],
+      '#options' => array(
+        'attributes' => array(
+          'title' => $variables['title'],
+        ),
+        'html' => FALSE,
+      ),
+      '#prefix' => '<h2>',
+      '#suffix' => '</h2>',
+    );
+  }
+
+  // Add updated to variables.
+  $variables['ddbasic_updated'] = t('!datetime', array(
+    '!datetime' => format_date(
+      $variables['node']->changed,
+      $type = 'long',
+      $format = '',
+      $timezone = NULL,
+      $langcode = NULL
+    ),
+  ));
+
+  // Modified submitted variable.
+  if ($variables['display_submitted']) {
+    $variables['submitted'] = t('!datetime', array(
+      '!datetime' => format_date(
+        $variables['created'],
+        $type = 'long',
+        $format = '',
+        $timezone = NULL,
+        $langcode = NULL
+      ),
+    ));
+  }
+}
+
+/**
+ * Ding news.
+ */
+function ddbasic_preprocess__node__ding_news(&$variables) {
+
+  $variables['news_submitted'] = format_date($variables['created'], 'ding_date_only_version2');
+  $variables['news_full_submitted'] = format_date($variables['created'], 'ding_date_and_time');
+  $variables['news_full_changed'] = format_date($variables['changed'], 'ding_date_and_time');
+  switch ($variables['view_mode']) {
+    case 'full':
+      array_push($variables['classes_array'], 'node-full');
+
+      // Make social-share button.
+      $share = '
+        <div class="social-share-container">
+          <div class="inner">
+            <div class="label">' . t('Share this news') . '</div>
+            <div class="share-buttons">
+              <a href="#" class="fb-share">Facebook</a>
+              <a href="#" class="twitter-share">Twitter</a>
+              <a href="#" class="mail-share">E-mail</a>
+            </div>
+          </div>
+        </div>
+      ';
+      $share_button = array(
+        '#theme' => 'field',
+        '#selected_theme_hook' => 'simple',
+        '#field_name' => 'share_button',
+        '#field_type' => 'fake',
+        '#label_display' => 'hidden',
+        '#bundle' => 'fake',
+        '#items' => array('0' => $share),
+        '0' => array('#markup' => $share),
+        '#weight' => 1,
+      );
+      $variables['content']['group_right']['share_button'] = $share_button;
```

Comment:
This seems wrong. Markup is not abstracted by a template. It is then added as a field.